test/widgets/subscription_list_test.dart

```diff
@@ -89,18 +89,71 @@ void main() {
     check(isUnpinnedHeaderInTree()).isTrue();
   });
 
-  testWidgets('subscription sort', (tester) async {
+  testWidgets('pinned subscriptions are sorted', (tester) async {
     await setupStreamListPage(tester, subscriptions: [
       eg.subscription(eg.stream(streamId: 1, name: 'd'), pinToTop: true),
-      eg.subscription(eg.stream(streamId: 2, name: 'c'), pinToTop: false),
+      eg.subscription(eg.stream(streamId: 2, name: 'c'), pinToTop: true),
       eg.subscription(eg.stream(streamId: 3, name: 'b'), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 4, name: 'a'), pinToTop: true),
+    ]);
+    check(isPinnedHeaderInTree()).isTrue();
+
+    final streamListItems = tester.widgetList<SubscriptionItem>(find.byType(SubscriptionItem)).toList();
+    check(streamListItems.map((e) => e.subscription.streamId)).deepEquals([4, 3, 2, 1]);
+  });
+
+  testWidgets('pinned subscriptions are sorted case insensitively', (tester) async {
+    await setupStreamListPage(tester, subscriptions: [
+      eg.subscription(eg.stream(streamId: 1, name: 'D'), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 2, name: 'c'), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 3, name: 'B'), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 4, name: 'a'), pinToTop: true),
+    ]);
```

Comment:
The case-insensitive part might be easiest to read if it's consolidated as one test case with four items: two pinned, and two unpinned, in order to exercise both.

(Or I guess six items, given the other point I mentioned.)

Reply:
Yes sure, additionally if I am getting it right, as we tested before that:
* pinned comes before unpinned streams
* both pinned and unpinned streams are sorted
I believe there will be no need to shuffle the streams this time as that would not be relevant to what we test (**that letter case does not matter in stream name sorting**) and that's why we refactored the tests in the first place. 
so something like this:
```dart
    testWidgets('subscriptions sorting is case insensitive', (tester) async {
      await setupStreamListPage(tester, subscriptions: [
        eg.subscription(eg.stream(streamId: 1, name: 'a'), pinToTop: true),
        eg.subscription(eg.stream(streamId: 2, name: 'B'), pinToTop: true),
        eg.subscription(eg.stream(streamId: 3, name: 'c'), pinToTop: true),
        eg.subscription(eg.stream(streamId: 4, name: 'D'), pinToTop: false),
        eg.subscription(eg.stream(streamId: 5, name: 'e'), pinToTop: false),
        eg.subscription(eg.stream(streamId: 6, name: 'F'), pinToTop: false),
      ]);

      check(streamListIds(tester)).deepEquals([1, 2, 3, 4, 5, 6]);
    });
```
Should be sufficient.

Reply:
Yes sure, additionally if I am getting it right, as we tested before that:
* pinned comes before unpinned streams
* both pinned and unpinned streams are sorted
I believe there will be no need to shuffle the streams this time as that would not be relevant to what we test (**that letter case does not matter in stream name sorting**) and that's why we refactored the tests in the first place. 
so something like this:
```dart
    testWidgets('subscriptions sorting is case insensitive', (tester) async {
      await setupStreamListPage(tester, subscriptions: [
        eg.subscription(eg.stream(streamId: 1, name: 'a'), pinToTop: true),
        eg.subscription(eg.stream(streamId: 2, name: 'B'), pinToTop: true),
        eg.subscription(eg.stream(streamId: 3, name: 'c'), pinToTop: true),
        eg.subscription(eg.stream(streamId: 4, name: 'D'), pinToTop: false),
        eg.subscription(eg.stream(streamId: 5, name: 'e'), pinToTop: false),
        eg.subscription(eg.stream(streamId: 6, name: 'F'), pinToTop: false),
      ]);

      check(streamListIds(tester)).deepEquals([1, 2, 3, 4, 5, 6]);
    });
```
Should be sufficient.

Reply:
Yes sure, additionally if I am getting it right, as we tested before that:
* pinned comes before unpinned streams
* both pinned and unpinned streams are sorted
I believe there will be no need to shuffle the streams this time as that would not be relevant to what we test (**that letter case does not matter in stream name sorting**) and that's why we refactored the tests in the first place. 
so something like this:
```dart
    testWidgets('subscriptions sorting is case insensitive', (tester) async {
      await setupStreamListPage(tester, subscriptions: [
        eg.subscription(eg.stream(streamId: 1, name: 'a'), pinToTop: true),
        eg.subscription(eg.stream(streamId: 2, name: 'B'), pinToTop: true),
        eg.subscription(eg.stream(streamId: 3, name: 'c'), pinToTop: true),
        eg.subscription(eg.stream(streamId: 4, name: 'D'), pinToTop: false),
        eg.subscription(eg.stream(streamId: 5, name: 'e'), pinToTop: false),
        eg.subscription(eg.stream(streamId: 6, name: 'F'), pinToTop: false),
      ]);

      check(streamListIds(tester)).deepEquals([1, 2, 3, 4, 5, 6]);
    });
```
Should be sufficient.